### ISSUE TITLE
[Typescript]: decorators cannot be generic

### ISSUE THREAD

**Issue by Lodin** [babel/babel#8753]
## Bug Report

**Current Behavior**
Trying to use generic decorators causes errors for all types of decorators (I tried for classes and properties). For classes it raises following error
```
SyntaxError: Leading decorators must be attached to a class declaration
```
For properties it complains about unexpected token.

**Input Code**
- REPL: https://runkit.com/lodin/babel-issue-typescript-decorators-generics 

```js
@decorator<string>()
class A {
  prop2 = 3;

  method2() {
    return this.prop2;
  }
}
```
**Expected behavior/code**
No error should be raised, generics should be removed by `@babel/plugin-transform-typescript`.

**Babel Configuration (.babelrc, package.json, cli command)**

```js
{
  plugins: [
    [require("@babel/plugin-proposal-decorators"), {decoratorsBeforeExport: true}],
    require("@babel/plugin-proposal-class-properties"),
  ],
  presets: [
    require("@babel/preset-typescript"),
  ],
  filename: "test.ts",
}
```

**Environment**
- Babel version(s): `7.1.0`
- Node/npm version: `v10.10.0`/`6.1.0`
- OS: `Windows 10`
- Monorepo: `Lerna`
- How you are using Babel: `loader`

**Comment by nicolo-ribaudo** [comment]
(NOTE: TypeScript only supports legacy decorators)

**Comment by Lodin** [comment]
@nicolo-ribaudo, I know and that's why I used the `@babel/preset-typescript`. Since there is almost no difference between decorator signatures, typescript compiler can be easily deceived and works perfectly well as a typechecker along with `@babel/preset-typescript` and `@babel/plugin-proposal-decorators`. 

The only problem I found is that I mentioned above. Looks like `@babel/preset-typescript` should remove generic types from decorators before it comes to decorator compilation with `@babel/plugin-proposal-decorators`.
